cmd/slinky/main.go

```diff
@@ -109,18 +126,12 @@ func runOracle() error {
 		}
 	}
 
-	var logger *zap.Logger
-	if !cfg.Production {
-		logger, err = zap.NewDevelopment()
-		if err != nil {
-			return fmt.Errorf("failed to create logger: %s", err.Error())
-		}
-	} else {
-		logger, err = zap.NewProduction()
-		if err != nil {
-			return fmt.Errorf("failed to create logger: %s", err.Error())
-		}
-	}
+	// Build logger.
+	logger := log.NewLogger(log.Config{
+		LogLevel: logLevel,
+		WriteTo:  writeLogsTo,
+	})
+	defer logger.Sync()
```

Comment:
What does this do?

Reply:
blocks on process closing till all logs are written